lib/Sema/TypeChecker.cpp

```diff
@@ -731,6 +732,7 @@ void swift::performWholeModuleTypeChecking(SourceFile &SF) {
   Ctx.diagnoseObjCMethodConflicts(SF);
   Ctx.diagnoseObjCUnsatisfiedOptReqConflicts(SF);
   Ctx.diagnoseUnintendedObjCMethodOverrides(SF);
+  performTopLevelDeclDiagnostics(SF);
```

Comment:
Good catch! I actually only put it there because I was originally exploring cross-file global variable checking, which turned out to be more complicated. I logged SR-3721 to track that. I'll move this to a better place.